core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/dr/BackupRestoreV4RequestValidator.java

```diff
@@ -1,64 +1,74 @@
+            AZURE, Pattern.compile("^abfs[s]?$", Pattern.CASE_INSENSITIVE),
+            GCP, Pattern.compile("^gs$", Pattern.CASE_INSENSITIVE)
+    );
 
     public ValidationResult validate(Stack stack, String location, String backupId) {
         ValidationResult.ValidationResultBuilder resultBuilder = ValidationResult.builder();
         String cloudPlatform = stack.cloudPlatform();
 
+        String validCloudPlatform = validateCloudPlatformAndReturn(cloudPlatform, resultBuilder);
         if (Strings.isNullOrEmpty(backupId)) {
             resultBuilder.error("Parameter backupId required");
         }
         if (Strings.isNullOrEmpty(location)) {
             resultBuilder.error("Parameter backupLocation required");
         } else {
-            validateCloudLocationScheme(cloudPlatform, location, resultBuilder);
+            validateCloudLocationScheme(validCloudPlatform, location, resultBuilder);
         }
         return resultBuilder.build();
     }
 
-    private void validateCloudLocationScheme(String cloudPlatform, String location, ValidationResult.ValidationResultBuilder resultBuilder) {
+    private String validateCloudPlatformAndReturn(String cloudPlatform, ValidationResult.ValidationResultBuilder resultBuilder) {
+        String validCloudPlatform = Arrays.stream(SUPPORTED_CLOUD_PLATFORM).filter(cloudPlatform::equalsIgnoreCase)
+                .findFirst().orElse(null);
+
+        if (validCloudPlatform == null) {
+            resultBuilder.error("Cloud platform " + cloudPlatform + " not supported for backup/restore");
+        }
+
+        return validCloudPlatform;
+    }
+
+    private void validateCloudLocationScheme(String validCloudPlatform, String location, ValidationResult.ValidationResultBuilder resultBuilder) {
         try {
             URI uri = new URI(location);
-            if (!Strings.isNullOrEmpty(uri.getScheme()) &&
-                    uri.getScheme().equalsIgnoreCase("hdfs")) {
+            if (Strings.isNullOrEmpty(uri.getScheme())) {
                 return;
             }
-            if (AWS.equalsIgnoreCase(cloudPlatform)) {
-                if (uri.getScheme() != null) {
-                    Matcher matcher = AWS_PATTERN.matcher(uri.getScheme());
-                    if (!matcher.find()) {
-                        resultBuilder.error("Incorrect URL scheme for AWS cloud platform: " + location);
-                    }
-                }
-            } else if (AZURE.equalsIgnoreCase(cloudPlatform)) {
-                if (uri.getScheme() != null) {
-                    Matcher matcher = AZURE_PATTERN.matcher(uri.getScheme());
-                    if (!matcher.find()) {
-                        resultBuilder.error("Incorrect URL scheme for Azure cloud platform: " + location);
-                    }
+            if (uri.getScheme().equalsIgnoreCase("hdfs")) {
+                return;
+            }
+            if (validCloudPlatform != null) {
```

Comment:
Similarly, I don't think we get here with validCloudPlatform being null, but we could keep an else case that also still throws an error